src/KimeraVioRos.cpp

```diff
@@ -121,15 +121,15 @@ bool KimeraVioRos::spin() {
     std::future<bool> data_provider_handle =
         std::async(std::launch::async,
                    &VIO::RosDataProviderInterface::spin,
-                   std::ref(*CHECK_NOTNULL(data_provider_.get())));
+                   CHECK_NOTNULL(data_provider_.get()));
     std::future<bool> vio_viz_handle =
         std::async(std::launch::async,
                    &VIO::Pipeline::spinViz,
-                   std::ref(*CHECK_NOTNULL(vio_pipeline_.get())));
+                   CHECK_NOTNULL(vio_pipeline_.get()));
     std::future<bool> vio_pipeline_handle =
         std::async(std::launch::async,
                    &VIO::Pipeline::spin,
-                   std::ref(*CHECK_NOTNULL(vio_pipeline_.get())));
+                   *CHECK_NOTNULL(vio_pipeline_.get()));
```

Comment:
lgtm: why do you need "*" here?

Reply:
`CHECK_NOTNULL(x)` returns the pointer `x` (if not null), so you still need to dereference it.

Reply:
got it thanks. I'll merge after Marcus reviews